src/configure/configure.ts

```diff
@@ -443,16 +443,41 @@ class Orchestrator {
         //Post selecting the template update this.inputs.repositoryAnalysisApplicationSettings with corresponding languageSettings
         if (!!repoAnalysisResult
             && !!repoAnalysisResult.repositoryAnalysisApplicationSettingsList) {
-
+            
             //Get languageSettings (corresponding to language of selected settings) provided by RepoAnalysis
-            this.inputs.repositoryAnalysisApplicationSettings = repoAnalysisResult.repositoryAnalysisApplicationSettingsList.find(applicationSettings => {
-                return applicationSettings.language === this.inputs.pipelineConfiguration.template.language;
-            });
+            this.updateRepositoryAnalysisApplicationSettings(repoAnalysisResult);
         }
 
         telemetryHelper.setTelemetry(TelemetryKeys.ChosenTemplate, this.inputs.pipelineConfiguration.template.label);
     }
 
+    private async updateRepositoryAnalysisApplicationSettings(repoAnalysisResult: RepositoryAnalysisParameters): Promise<void>{
+        var applicationSettings = repoAnalysisResult.repositoryAnalysisApplicationSettingsList.filter(applicationSetting => {
+            return applicationSetting.language === this.inputs.pipelineConfiguration.template.language;
+        });
+        
+        let workspacePaths = Array.from(new Set(applicationSettings.map(a => a.settings.workingDirectory)));
+        if(workspacePaths.length == 1){
+            this.inputs.repositoryAnalysisApplicationSettings = applicationSettings[0];
+            this.inputs.pipelineConfiguration.workingDirectory = applicationSettings[0].settings.workingDirectory;
+            return;
+        }
+
+        let workspacePathQuickPickItemList: Array<QuickPickItemWithData> = [];
+        for (let workspacePath of workspacePaths) {
+            workspacePathQuickPickItemList.push({ label: workspacePath, data: workspacePath });
+        }
+        let selectedWorkspacePathItem = await this.controlProvider.showQuickPick(
+            constants.SelectWorkspace,
+            workspacePathQuickPickItemList,
+            { placeHolder: Messages.selectWorkspace });
+        
+        this.inputs.pipelineConfiguration.workingDirectory = selectedWorkspacePathItem.data;
+        this.inputs.repositoryAnalysisApplicationSettings = repoAnalysisResult.repositoryAnalysisApplicationSettingsList.find(applicationSettings => {
```

Comment:
based on the logic of our repo analysis, this filter can yield more than one elements - think of case where there are two gruntfiles under same working directory (package.json). In such cases there will be elements in repoAnalysisResult which will match the filter condition.
We will have to drill down further and ask user to select specific gulp/grunt or other such file to zero down on one element

Reply:
No, that is why I've used HashSet above to fetch unique list of working directories